client/components/button/docs/example.jsx

```diff
@@ -59,7 +59,8 @@ class Buttons extends React.PureComponent {
 					<div className="docs__design-button-row">
 						<Button>Button</Button>
 						<Button>
-							<Gridicon icon="heart" /> Icon button
+							<Gridicon icon="heart" />
```

Comment:
I think we should remove the span for all of these examples, and let the developer decide if they need the extra span around the text.

Reply:
You mean remove all, or show in the examples some with and some without? (I'd do the second)

Reply:
Without the `span`, there won't be the gap between the icon and the text.

For me, this PR has two purposes:
1. To remove the unwanted gap in an icon only buttons.
2. More importantly, to establish what's the prefered way to give a gap between the icon and the text. (until we rework the component so that we can specify an icon as a prop)

My opinion on the second point is that we shouldn't have to add the gap every time to the icon, but instead, wrap the text with something like a span, and the component CSS takes care of that. If this was our consensus the devdoc should show that, shouldn't it?

Reply:
Ah yes... I have misunderstood, I was considering there were scenarios when that wasn't the case, but you're right. The example should show how it works, so always wrapping.